sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClientBuilder.java

```diff
@@ -763,10 +857,43 @@ public EventProcessorClient buildEventProcessorClient() {
             partitionOwnershipExpirationInterval = DEFAULT_OWNERSHIP_EXPIRATION_INTERVAL;
         }
 
-        return new EventProcessorClient(eventHubClientBuilder, consumerGroup,
-            getPartitionProcessorSupplier(), checkpointStore, trackLastEnqueuedEventProperties,
-            processError, initialPartitionEventPosition, maxBatchSize, maxWaitTime, processEventBatch != null,
-            loadBalancingUpdateInterval, partitionOwnershipExpirationInterval, loadBalancingStrategy, eventHubClientBuilder.createTracer());
+        final EventProcessorClientOptions processorOptions = new EventProcessorClientOptions()
+            .setConsumerGroup(consumerGroup)
+            .setBatchReceiveMode(processEventBatch != null)
+            .setConsumerGroup(consumerGroup)
+            .setLoadBalancingStrategy(loadBalancingStrategy)
+            .setLoadBalancerUpdateInterval(loadBalancingUpdateInterval)
+            .setMaxBatchSize(maxBatchSize)
+            .setMaxWaitTime(maxWaitTime)
+            .setPartitionOwnershipExpirationInterval(partitionOwnershipExpirationInterval)
+            .setTrackLastEnqueuedEventProperties(trackLastEnqueuedEventProperties);
+
+        int numberOfTimesSet = 0;
+
+        if (defaultInitialEventPosition != null) {
+            numberOfTimesSet++;
+            processorOptions.setInitialEventPositionProvider(unused -> defaultInitialEventPosition);
+        }
+
+        if (initialPartitionEventPosition != null) {
+            numberOfTimesSet++;
+            processorOptions.setInitialEventPositionProvider(
+                partitionId -> initialPartitionEventPosition.get(partitionId));
+        }
+
+        if (initialEventPositionProvider != null) {
+            numberOfTimesSet++;
+            processorOptions.setInitialEventPositionProvider(initialEventPositionProvider);
+        }
+
+        if (numberOfTimesSet > 1) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                "Only 1 overload for setInitialEventPositionProvider can be set.  The overload is set "
```

Comment:
nit;  just confirming the message, do we mean "Only 1 overload of 'initialPartitionEventPosition' can be set ..." given that's the overloaded public api name?

Reply:
THanks. I will fix that.